indigo_social/views.py

```diff
@@ -292,3 +294,19 @@ class UserPopupView(AbstractAuthedIndigoView, DetailView):
     slug_url_kwarg = 'username'
     template_name = 'indigo_social/user_popup.html'
     queryset = User.objects
+
+
+class UserProfilePhotoView(AbstractAuthedIndigoView, View):
+    def get(self, request, **kwargs):
+        username = kwargs['username']
+        nonce = kwargs['nonce']
+
+        user = get_object_or_404(User, username=username)
+
+        if not user.userprofile.profile_photo:
+            return redirect(static('images/avatars/default_avatar.svg'))
+
+        response = HttpResponse(user.userprofile.profile_photo.read())
+        response['Content-Disposition'] = 'inline; filename={}'.format(user.userprofile.profile_photo.name)
```

Comment:
This header isn't necessary, since we don't really care about the user downloading the file to the filesystem.